### DISCUSSION TITLE
How can I setup `/metrics` handler to expose the default metrics collected by ogen server?

### DISCUSSION THREAD

**Discussion by zwpaper** [ogen-go/ogen#1239] [Q&A] [answered]
I noticed there is an example with otel support, `https://github.com/ogen-go/example`, but it seems to be complicated, and depends on `https://github.com/go-faster/sdk`, I just want to expose the metrics collected by ogen server by default

**Comment by ghost** [comment, marked as answer]
You can create a Prometheus `MeterProvider` by yourself: 

```go
exporter, err := prometheus.New()
if err != nil {
	return err
}
provider := metric.NewMeterProvider(metric.WithReader(exporter))

srv, err := <api_package>.NewServer(<api_package>.WithMeterProvider(provider))
if err != nil {
	return err
}

http.Handle("/metrics", promhttp.Handler())
http.Handle("/api", http.StripPrefix("/api", srv))
http.ListenAndServe(":8080", nil)
```

There is OpenTelemery SDK [example](https://github.com/open-telemetry/opentelemetry-go/blob/main/example/prometheus/main.go).

**Reply by zwpaper** [reply]
`http.Handle("/api", http.StripPrefix("/api", srv))` should be like:

```go
http.Handle("/api/", http.StripPrefix("/api", srv))
```

to catch all the `/api/` routes
